shell/platform/linux/fl_key_event_plugin.h

```diff
@@ -27,23 +27,35 @@ G_DECLARE_FINAL_TYPE(FlKeyEventPlugin,
 /**
  * fl_key_event_plugin_new:
  * @messenger: an #FlBinaryMessenger.
+ * @response_callback: the callback to call when a response is received.  If not
+ *                     given (nullptr), then the default response callback is
+ *                     used.
+ * @channel_name: the name of the channel to send key events on. If not given
+ *                (nullptr), then the standard key event channel name is used.
+ *                Typically used for tests to send on a test channel.
  *
  * Creates a new plugin that implements SystemChannels.keyEvent from the
  * Flutter services library.
  *
  * Returns: a new #FlKeyEventPlugin.
  */
-FlKeyEventPlugin* fl_key_event_plugin_new(FlBinaryMessenger* messenger);
+FlKeyEventPlugin* fl_key_event_plugin_new(
+    FlBinaryMessenger* messenger,
+    GAsyncReadyCallback response_callback = nullptr,
```

Comment:
Do we use `response_callback` and `user_data` anywhere? I might have missed. If we don't, then shall we not add them for now?

Reply:
I use them to inject a callback and channel name ("test/echo") for testing.

Reply:
Cool, but why are the test changes not in this PR?

Reply:
Yeah, I just realized that I forgot to add the file.  Its in now.